src/messages/mod.rs

```diff
@@ -403,13 +421,29 @@ impl SignedRoutingMessage {
             SecurityMetadata::None | SecurityMetadata::Partial(_) => {
                 Err(RoutingError::FailedSignature)
             }
-            SecurityMetadata::Single(security_metadata) => security_metadata.verify(&self.content),
+            SecurityMetadata::Single(security_metadata) => security_metadata
+                .verify(&self.content)
+                .map(|_| VerifyStatus::Full),
             SecurityMetadata::Full(security_metadata) => {
                 security_metadata.verify(&self.content, their_key_infos)
             }
         }
     }
 
+    /// Verifies that the proof and the signature of this message are valid, but does not check
+    /// the proof against known keys.
+    pub fn verify_partial(&self) -> Result<(), RoutingError> {
```

Comment:
I'm not sure why we want verify partial. Essentially with my changeset you may not have a public key, so it is not possible.

I'd say, maybe if you want to check the signature you could look if there is a block and check against it, but you would need to make sure they give at least one block for these new node otherwise that would amount to nothing.

But should they really do that? we still don't have any trust root so why trust the key they give us?

Reply:
I guess checking at least that the proof is valid (each block is signed with the previous block) and the signature is valid is better than nothing? But maybe it isn't, because without trusted keys, a malicious agent can forge any proofs/signatures without much effort. So probably better to not verify the message at all for now?